src/components/QuizSettings/index.tsx

```diff
@@ -40,50 +41,44 @@ const useStyles = makeStyles({
   }
 });
 
-// TODO: add error handling
+// TODO: add form validation
 export const QuizSettingsForm = ({
   newUserSetting,
   setSendRequest,
 }: SettingPropsType) => {
-  const [categories, setCategories] = useState<CategoriesType[]>([])
   const [newSetting, setUserSetting] = useState<SettingType>({
-      numberOfQuestions: 5,
-      difficulty: Difficulty.EASY,
-      category: 9,
-      categoryName: "General Knowledge",
-      name: "",
-    });
-
-    const classes = useStyles()
-
-    useEffect(() => {
-      const getCategoriesData = async () => {
-        const fetchedCategories = await fetchQuizCategories()
-        setCategories(fetchedCategories)
-      };
+    numberOfQuestions: 5,
+    difficulty: Difficulty.EASY,
+    category: 9,
+    categoryName: "General Knowledge",
+    name: "",
+  });
   
-      getCategoriesData()
-    }, []);
-  
-    if (!categories.length) {
-      return (
-          <div className="loading">
-              ...loading
-          </div>
-      );
-    }
-  
-    let categoryName = categories.filter((category) => {
-        return category.id === newSetting.category
-    });
+  const classes = useStyles()
+  const { data } = useQuery<CategoryResponse, Error, CategoriesType[]>(
+    ["categories"], 
+    useQuizCategories
+  );
```

Comment:
ℹ️ This is what this could look like if we moved the `useQuery` hook into your custom `useQuizCategories` hook...
```suggestion
  const { categories } = useQuizCategories();
```